### DISCUSSION TITLE
How to handle errors in Swagger UI for api endpoints?

### DISCUSSION THREAD

**Discussion by Jamadan** [jellydn/next-swagger-doc#651] [Q&A]
I built this - https://github.com/Jamadan/swag-test because I'm tryng to integrate next-swagger-doc into a next app I'm working on

You can clone, `yarn`, `yarn openapi:yaml`, `yarn dev`, then navigate to `localhost:3000/api-doc` and I'm hoping for some input here

First issue - I'm seeing UNSAFE_componentWillReceiveProps error on page load. Is there anything I can about this or is this something to do with Swagger UI and how it is building based on the `pages/api` directory?

<img width="1552" alt="next-swagger-doc-1" src="https://user-images.githubusercontent.com/3406219/227367924-43ba7c33-4067-44f8-bf0e-637e8d29e657.png">

Second issue - whats the correct way of handling the swagger validation errors? I'm getting an error in my console that the required value is not supplied, but the page hangs and I don't see anything in my network tab. I'd like to return a 400 bad request and show this in the Swag UI, but not sure of the correct way to do this, or where to put an error handler or middlewear. I could make the query params not required and do some check in the function to do this, but thought I would get some validation out of Swag UI here

<img width="1532" alt="next-swagger-doc-2" src="https://user-images.githubusercontent.com/3406219/227368751-ae055d97-d13c-4b0a-ae47-b06746724c56.png">

Third issue - if I correctly execute with values, I get the 200, however, following that with undefined values, sees the console logs found here https://github.com/Jamadan/swag-test/blob/main/src/pages/api/hello.ts#L32 print `undefined, undefined`. So again  could manually check the values and `res.status(400)` but thought the swagger validation would do this?

Any thoughts/help would be appreciated!

Thanks

**Comment by jellydn** [comment]
Hi @Jamadan 

> UNSAFE_componentWillReceiveProps

That's warning from swagger UI component. I think you should open issue for them to migrate/fix this warning for React.

> whats the correct way of handling the swagger validation errors?

I would recommend that you generate swagger from CLI as I've mentioned on the README. Then you could verify the schema if any issue with other tools (e.g: Swagger Editor Online or VSCode extension would be useful)

> Third issue - if I correctly execute with values, I get the 200, however, following that with undefined values, sees the console logs found here https://github.com/Jamadan/swag-test/blob/main/src/pages/api/hello.ts#L32 print undefined, undefined. So again could manually check the values and res.status(400) but thought the swagger validation would do this?

I'm not so sure about this one. Let's see if anyone know the answer.

**Reply by Jamadan** [reply]
Thanks @jellydn - I'll raise with Swagger the first one, and will follow instructions on CLI generation. Appreciate the input
